doc/source/whatsnew/v0.25.0.rst

```diff
@@ -325,7 +325,7 @@ Indexing
 ^^^^^^^^
 
 - Improved exception message when calling :meth:`DataFrame.iloc` with a list of non-numeric objects (:issue:`25753`).
--
+- Bug in which :meth:`DataFrame.append` produced a warning indicating that a KeyError will be thrown in the future when the data to be appended contains new columns (:issue:`22252`).
```

Comment:
double backticks on KeyError. make this clear that this warning should not have been shown.